### PR TITLE
NIAD-169 Interchange-OperationId Reports and other documentation improvements

### LINKED ISSUE #219: NHAIS_MESH_HOST must have a trailing slash in order to work
This is a minor issue. 

Apparently, NHAIS_MESH_HOST variable currently must have a trailing slash at the end in order to work. 

suggestion
(1) should be captured in the documentation of the variable.
(2) or recoding to accept both with/without trailing slash.

### LINKED ISSUE #225: Documentation about the HA Cipher
The HACipher in the message and HACipher in the deployment configuration can be different because of this rule, where it has to be four characters long in the EDIFACT message

This needs to be documented particularly when setting this variable NHAIS_MESH_CYPHER_TO_MAILBOX.

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

File: REPORTING.md [hunk 1]
```diff
@@ -204,4 +211,59 @@ Report on interchanges that have not received RECEP from `2020-01-01` to `2020-0
...
+- `<from_interchange_sequence_number>` is the first sequence number in the range NHAIS reported as lost
+- `<to_interchange_sequence_number>` is the last sequence number in the range NHAIS reported as lost
+
+### Interchange-OperationId Report Example
+
+Searching for OperationIds for interchanges 3-10:
+
+    db.getCollection('outboundState').find(
+            {
+                $and: [
```
Comment (mattd-kainos) [inline REPORTING.md]: Need to add 'sndr' and 'recip' filters because these will apply to a single pair

File: REPORTING.md [hunk 2]
```diff
@@ -204,4 +211,59 @@ Report on interchanges that have not received RECEP from `2020-01-01` to `2020-0
...
+            {
+                intSeq: 1,
+                operationId: 1,
+                _id: 0
+            }
+     )
+     
+Produces two interchange sequence number / operation id pairs
+
+    /* 1 */
```
Comment (mattd-kainos) [inline REPORTING.md:270]: Why did my query not find `intSeq: 4`?

**Review by michael-kainos — DISMISSED** [review]

File: REPORTING.md [hunk 3]
```diff
@@ -38,14 +38,21 @@ A "Missing RECEP Report" can detect these faults.
 
 ## Missing Inbound Transaction
 
-It is possible that an inbound transaction was not processed successfully by the adaptor or lost in transmission. If any
-missing inbound transaction is identified the GP System operator must check the dead-letter queues for any messages that
+It is possible, however unlikely, that an inbound transaction fails to process successfully is lost in transmission. If any
```
Comment (Alicja-Kainos) [inline REPORTING.md:41]: failed/failing to process?

**Review by Alicja-Kainos — DISMISSED** [review]

File: REPORTING.md [hunk 4]
```diff
@@ -38,14 +38,21 @@ A "Missing RECEP Report" can detect these faults.
...
-are no matching messages in the dead-letter queues then the GP System operator must ask the NHAIS operator for the reason
-why the message was not received.
+are no matching messages in the dead-letter queues then the GP System operator must ask the NHAIS operator to investigate
+and possibly resend the lost message.
 
 A "Missing Sequence Number Report" on the `inboundState` collection can detect these faults.
 
+## Matching Interchange Sequence Numbers to Operation Ids
+
+It is possible, however unlikely, that the NHAIS operator may call upon the GP System to re-enter transactions due to
```
Comment (MattLewis80) [inline REPORTING.md]: two paragraphs starting 'It is possible, however unlikely, ' will stick out like a sore thumb.  Perhaps we can amend the second instance to 'It is also possible, that...'

**Review by MattLewis80 — DISMISSED** [review]

**Review by grzegorz-robak — APPROVED** [review]

**Pull request merged**